x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -364,17 +364,21 @@ private static void checkCategorizeGrouping(Aggregate agg, Set<Failure> failures
                 );
         });
 
-        // Forbid CATEGORIZE being used in the aggregations
-        agg.aggregates().forEach(a -> {
-            a.forEachDown(
-                Categorize.class,
-                categorize -> failures.add(
-                    fail(categorize, "cannot use CATEGORIZE grouping function [{}] within the aggregations", categorize.sourceText())
+        // Forbid CATEGORIZE being used in the aggregations, unless it appears as a grouping
+        agg.aggregates()
+            .forEach(
+                a -> a.forEachDown(
+                    AggregateFunction.class,
+                    aggregateFunction -> aggregateFunction.forEachDown(
+                        Categorize.class,
+                        categorize -> failures.add(
+                            fail(categorize, "cannot use CATEGORIZE grouping function [{}] within an aggregation", categorize.sourceText())
```

Comment:
Is this specific to CATEGORIZE? Or should this apply to all grouping functions?

Reply:
Only categorize; BUCKET does work, as it works technically as a normal scalar function.
I guess, if another stateful grouping function appears, we may either add a `StatefulGroupingFunction` superclass, or change Bucket to be a normal function with exceptions (maybe?)